### PR TITLE
layers: install headers in vulkan subdirectory

### PR DESCRIPTION
**Pull request by djdeath**
Addressing #729 

Signed-off-by: Lionel Landwerlin <lionel.g.landwerlin@intel.com>

### CODE REVIEW

File: layers/CMakeLists.txt [hunk 1]
```diff
@@ -81,7 +81,7 @@ set(LAYER_UTIL_FILES
     ${PROJECT_BINARY_DIR}/vk_object_types.h
     ${PROJECT_BINARY_DIR}/vk_extension_helper.h
     ${PROJECT_BINARY_DIR}/vk_typemap_helper.h)
-install(FILES ${LAYER_UTIL_FILES} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(FILES ${LAYER_UTIL_FILES} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/vulkan)
```
Comment (jzulauf-lunarg) [inline layers/CMakeLists.txt]: vk_layer.h isn't even on this list.  most of these files are internal to the build of the validation layers.

Comment (djdeath) [reply layers/CMakeLists.txt]: vk_layer.h is installed by Vulkan-Headers.
It's all the headers in the list we're having issues with.

**Comment by mikew-lunarg** [comment]
@djdeath May I request a brief summary of the rationale for this change in the commit's description, to be captured in the repo's history.  Thanks.

**Comment by djdeath** [comment]
Sure.

The header repository installs headers in $CMAKE_INSTALL_INCLUDEDIR/vulkan 
 : https://github.com/KhronosGroup/Vulkan-Headers/blob/master/CMakeLists.txt#L39
The basically leaves no choice to the user, forcing the `vulkan` subdirectory to be created.
The validation layers don't have that same behavior and it seems that a few Linux distribution choice a different path that makes it impossible for applications or custom layers to find the include path for the headers.

**Comment by djdeath** [comment]
Commit description updated.

**Comment by mikew-lunarg** [comment]
@djdeath Thanks for the PR.  I think it's clear from the TravisCI failure that this 1-line approach will be more invasive than expected.  You've brought up a reasonable use-model, and I believe we need to define a actual specification for which and where headers should be installed for public access; not all of the files in  LAYER_UTIL_FILES really should be copied.  

I've created a new issue for that effort, see #743 and closing this PR.  Would appreciate your input there, esp re the files you need to access for what you're trying to accomplish.

**Pull request closed without merging**